src/Translator.php

```diff
@@ -119,10 +119,20 @@ private function scanDir(string $dir): array
         return array_diff(scanDir($dir), ['.', '..']);
     }
 
+    private function getFrameworkMajor(): string
+    {
+        $output = $this->exec('composer show silverstripe/framework | grep versions');
+        if (preg_match('#^versions : \* ([4-9])#', $output, $matches)) {
+            return $matches[1];
+        }
+        return '5';
```

Comment:
We should throw a `LogicException` if we don't get a match - it indicates that we've done something wrong either with the setup of our environment or in the code for the translator.

Reply:
Updated